spec/services/search/chat_channel_membership_spec.rb

```diff
@@ -218,5 +218,30 @@ def index_documents(resources)
       chat_channel_membership_docs = described_class.search_documents(params: params, user_id: user.id)
       expect(chat_channel_membership_docs.count).to eq(1)
     end
+
+    it "paginates the results" do
+      allow(chat_channel_membership1).to receive(:channel_last_message_at).and_return(Time.current)
+      allow(chat_channel_membership2).to receive(:channel_last_message_at).and_return(1.year.ago)
+      index_documents([chat_channel_membership1, chat_channel_membership2])
+      first_page_params = { page: 0, per_page: 1, sort_by: "channel_last_message_at", order: "dsc" }
+
+      chat_channel_membership_docs = described_class.search_documents(params: first_page_params, user_id: user.id)
+      expect(chat_channel_membership_docs.first["id"]).to eq(chat_channel_membership1.id)
+
+      second_page_params = { page: 1, per_page: 1, sort_by: "channel_last_message_at", order: "dsc" }
+
+      chat_channel_membership_docs = described_class.search_documents(params: second_page_params, user_id: user.id)
+      expect(chat_channel_membership_docs.first["id"]).to eq(chat_channel_membership2.id)
+    end
+
+    it "returns an empty Array if no results are found" do
+      allow(chat_channel_membership1).to receive(:channel_last_message_at).and_return(Time.current)
+      allow(chat_channel_membership2).to receive(:channel_last_message_at).and_return(1.year.ago)
+      index_documents([chat_channel_membership1, chat_channel_membership2])
+      params = { page: 3, per_page: 1 }
+
+      chat_channel_membership_docs = described_class.search_documents(params: params, user_id: user.id)
+      expect(chat_channel_membership_docs.count).to eq(0)
```

Comment:
On the danger of sounding extremely nitpicky: the description of this spec says "returns an empty Array" so I would have sort of expected `expect(chat_channel_membership_docs).to eq([])` since the count of an empty hash for example would also be 0. 

No big deal though, definitely not enough to keep you from merging this 😃

Reply:
No, you're right! I actually had both before I got slapped with method too long. I must've chopped the wrong line by accident. I made a note to start a discussion about extending the limit from 10 lines to 15 so that'll be coming soon.